contractcourt/channel_arbitrator.go

```diff
@@ -197,6 +209,19 @@ type ContractReport struct {
 	RecoveredBalance btcutil.Amount
 }
 
+// resolverReport creates a resolve report using some of the information in the
+// contract report.
+func (c *ContractReport) resolverReport(spendTx *chainhash.Hash,
+	outcome channeldb.ResolverOutcome) *channeldb.ResolverReport {
+
+	return &channeldb.ResolverReport{
+		OutPoint:        c.Outpoint,
+		Amount:          c.Amount,
+		ResolverOutcome: outcome,
+		SpendTxID:       spendTx,
+	}
+}
```

Comment:
Can't you add `ResolverOutcome` and `SpendTxID` to `ContractReport`? Conceptually it makes sense to me to let every resolver only have one report structure for its current state.

Reply:
The `ContractReport` struct doesn't really align with what we need for the resolver level reporting. 
Specifically, our two stage contest resolvers report on their final htlcs/amounts, not the success/failure transactions that they are actually dealing with before the hand off the the timeout/claim stage. 

This would need a full overhaul of our current reports to set them up in these stages (not even sure we want that), and it would result in a behavior change on rpc. While these two reports have common fields, they don't actually align too well. We also don't gain a whole lot, because we set the values on the report then immediately checkpoint (the values don't need to be saved for later use).

Reply:
Had another look at this and it's not going to work. Would need to add locks for existing reports, change the meaning of our contest resolver reports and then decide whether to use the nursery report or arb report (because the nursery sometimes reports, sometimes doesn't). If we do need a pending channels overhaul, these can easily be unified later imo.

Reply:
The lock is indeed required, but not a new thing. `commitSweepResolver` already locks.

The decision logic to use the nursery or resolver report is already present in `rpcserver.go`. Would that need a lot of change then?

I thought the conclusion of our offline discussion was that the new resolver report could be mapped on the current pending report pretty easily. How exactly isn't that possible? Or maybe something is missing in the new report to allow this (also preventing an easy implementation in the future)?

Reply:
If we decide it's a priority to address pending channels gaps, we can identify these gaps first (I don't know what they are, and we don't have many issues concerning them) and then combine the two. There's nothing in this that prevents us from doing so, it's a matter of more code rather than different code.  We need adding locking/reports to the resolvers that don't have them, then updating rpc to only use reports (which means itest changes as well). It's just another 5/6 commits in an already large, long lived change to a sensitive part of the code base.